### PR TITLE
Remove peripheralId from SimulatedService

### LINKED ISSUE #276: Remove peripheralId from SimulatedService
There are already methods that we can use for finding desired characteristic in all available peripherals.

### PR DESCRIPTION
**Pull request by mikolak**
Resolves https://github.com/Polidea/FlutterBleLib/issues/276

### CODE REVIEW

**Comment by pawelByszewski** [comment]
What about starting introducing some unit tests in changed areas?

File: lib/internal/simulation_manager_mixins/characteristics_mixin.dart [hunk 1]
```diff
@@ -87,10 +87,10 @@ mixin CharacteristicsMixin on SimulationManagerBaseWithErrorChecks {
 
     await _errorIfCharacteristicIsNull(
         targetCharacteristic, characteristicUUID);
-    await _errorIfNotConnected(targetCharacteristic.service.peripheralId);
+    await _errorIfNotConnected(_peripheralWithServiceId(serviceIdentifier).id);
```
Comment (bartoszwilk) [inline lib/internal/simulation_manager_mixins/characteristics_mixin.dart]: _findPeripheralWithServiceId? _findPeripheralForServiceId? _getPeripheralForServiceId?

File: lib/internal/simulation_manager_mixins/characteristics_mixin.dart [hunk 2]
```diff
@@ -189,17 +190,26 @@ mixin CharacteristicsMixin on SimulationManagerBaseWithErrorChecks {
               _peripheralWithCharacteristicId(characteristicIdentifier).id);
 
           _bridge.publishCharacteristicUpdate(
-              targetCharacteristic, value, transactionId);
+              _peripheralWithCharacteristicId(characteristicIdentifier).id,
```
Comment (bartoszwilk) [inline lib/internal/simulation_manager_mixins/characteristics_mixin.dart]: _findPeripheralWithCharacteristicId? _findPeripheralForCharacteristicId? _getPeripheralForCharacteristicId?

**Review by bartoszwilk — DISMISSED** [review]

**Review by pawelByszewski — CHANGES_REQUESTED** [review]
What about starting introducing some unit tests in changed areas?

**Review by pawelByszewski — APPROVED** [review]

**Pull request merged**